student.rb

```diff
@@ -1,12 +1,16 @@
 require_relative './person'
 
 class Student < Person
-  def initialize(classroom)
+  def initialize(classroom, age, name = 'Unknown', parent_permission: true)
     super()
+    super(name, age, parent_permission)
     @classroom = classroom
   end
 
   def play_hooky
     "¯\(ツ)/¯"
   end
+
+  @classroom = classroom
+  classroom.students.push(self) if classroom.students.include?(self)
```

Comment:
- [ ] I think we should use `unless` instead of `if` on line 15 to avoid duplicate entries.

- [ ] I think this should be enclosed in a setter method, like so:

  ```
  def classroom=(classroom)
    @classroom = classroom
    classroom.students.push(self) unless classroom.students.include?(self)
  end
  ```